components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQInjectHandler.java

```diff
@@ -139,6 +139,14 @@ public boolean invoke(RabbitMQMessage message) {
             log.error("Sequence: " + injectingSeq + " not found");
         }
 
+        Object o = msgCtx.getProperty(RabbitMQConstants.SET_ROLLBACK_ONLY);
+        if (o != null) {
```

Comment:
You write the following logic as follows. In my view this is more readable

if ( o == null ) {
    return true;
}

return  !(o instanceof Boolean && (Boolean) o || (o instanceof String && Boolean.valueOf((String) o)));